### PR TITLE
improved coverage for package of proxy

### PR DESCRIPTION
**Pull request by sluk3r**
test case code added for classes of DatabaseCommunicationEngineFactoryTest and ProxyJDBCExecutePrepareCallbackTest

### CODE REVIEW

File: sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/DatabaseCommunicationEngineFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+    
+    @Test
+    public void assertNewBinaryProtocolInstance() {
+        DatabaseCommunicationEngine engine = DatabaseCommunicationEngineFactory.getInstance()
+                .newBinaryProtocolInstance(mock(LogicSchema.class), "schemaName", Collections.emptyList(), mock(BackendConnection.class));
+        assertNotNull(engine);
+        assertThat(engine, instanceOf(JDBCDatabaseCommunicationEngine.class));
+    }
+}
```
Comment (KomachiSion) [inline sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/DatabaseCommunicationEngineFactoryTest.java]: Files should be end with a empty line.

File: sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/execute/callback/ProxyJDBCExecutePrepareCallbackTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    @Test
+    public void assertCreateStatementExecuteUnitWhenMemoryStrictly() throws SQLException {
+        JDBCExecutorWrapper jdbcExecutorWrapper = mock(JDBCExecutorWrapper.class);
+        when(jdbcExecutorWrapper.createStatement((Connection) any(), (SQLUnit) any(), anyBoolean())).thenReturn(mock(Statement.class));
+        ProxyJDBCExecutePrepareCallback proxyJDBCExecutePrepareCallback = new ProxyJDBCExecutePrepareCallback(mock(BackendConnection.class), jdbcExecutorWrapper, false);
+        RouteUnit routeUnit = mock(RouteUnit.class);
+        when(routeUnit.getSqlUnit()).thenReturn(mock(SQLUnit.class));
+        assertThat(proxyJDBCExecutePrepareCallback.createStatementExecuteUnit(null, routeUnit, ConnectionMode.MEMORY_STRICTLY), instanceOf(StatementExecuteUnit.class));
+    }
+}
```
Comment (KomachiSion) [inline sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/execute/callback/ProxyJDBCExecutePrepareCallbackTest.java]: Files should be end with a empty line.

**Comment by KomachiSion** [comment]
One more thing is your commit reports, May you read the develop flow first?

[develop flow](https://shardingsphere.apache.org/community/en/contribute/contributor/)

File: sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/DatabaseCommunicationEngineFactoryTest.java (hunk 1 above)
Comment (sluk3r) [reply sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/DatabaseCommunicationEngineFactoryTest.java]: It seems that the last blank line is hidden in the changed files page. 

In IDE, the last blank lines is visible. 
image

And my merged pr's changed files page also does not show the last blank line, as seen in pr: https://github.com/apache/incubator-shardingsphere/pull/3204/files

File: sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/execute/callback/ProxyJDBCExecutePrepareCallbackTest.java (hunk 2 above)
Comment (sluk3r) [reply sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/execute/callback/ProxyJDBCExecutePrepareCallbackTest.java]: same as the above conversation.

**Pull request merged**